vmm/src/vm.rs

```diff
@@ -213,9 +213,20 @@ fn get_host_cpu_phys_bits() -> u8 {
     unsafe {
         let leaf = x86_64::__cpuid(0x8000_0000);
 
+        let reduced = if leaf.eax >= 0x8000001f
+            && leaf.ebx == 0x68747541
```

Comment:
From https://www.kernel.org/doc/Documentation/x86/amd-memory-encryption.txt
```
Support for SME and SEV can be determined through the CPUID instruction. The
CPUID function 0x8000001f reports information related to SME:

	0x8000001f[eax]:
		Bit[0] indicates support for SME
		Bit[1] indicates support for SEV
	0x8000001f[ebx]:
		Bits[5:0]  pagetable bit number used to activate memory
			   encryption
		Bits[11:6] reduction in physical address space, in bits, when
			   memory encryption is enabled (this only affects
			   system physical addresses, not guest physical
			   addresses)
```

I don't see any mention of ecx or edx.

Reply:
What I do here is checking the vendor string of amd.

0x68747541 Auth  0x444d4163 cAMD  0x69746e65 enti